app/views/users/_follow.html.erb

```diff
@@ -1,4 +1,10 @@
-<%= form_for(current_user.active_relationships.build) do |f| %>
-  <div><%= hidden_field_tag :followed_id, @profile_user.id %></div>
-  <%= f.submit "Follow", class: "btn btn-primary" %>
+<% if !current_user %>
+         <a data-toggle="modal" data-target="#loginModal"><div><%= hidden_field_tag :followed_id, @profile_user.id %></div></a>
+                <%= f.submit "Follow", class: "btn btn-primary" %>
+<% end %>
```

Comment:
Have you studied the if else and form_for helper?
Your code is not following their guidelines.
Please check

Reply:
ok I will. Also, should I be doing something similar to this:
https://github.com/okonek/plots2/blob/f8baa2adf72c8e8f82fb1cabcb7085e00355f987/app/views/like/_like.html.erb#L13

Reply:
@SidharthBansal like this? https://launchschool.com/books/demystifying_rails/read/form_for_helper